pandas/core/arrays/interval.py

```diff
@@ -284,7 +306,10 @@ def _simple_new(
         from pandas.core.indexes.base import ensure_index
 
         left = ensure_index(left, copy=copy)
```

Comment:
Is it possible to handle this somehow in the creation of the IntervalIndex?

Reply:
I tried to remove `ensure_index` in various ways, but failed. It looks like there is some dtype issues that need to passed through an Index to be solved, but I didn't manage to untangle it, unfortunately.

There is a lot going on in `IntervalArray._simple_new`. I've looked into moving all the validation/dtype wrangling there into a separate function. That would mean that `_simple_new` would become much more simple and it would much faseter/simpler to instantiate an `IntervalArray`, when we can be sure the input data is correct. I'll push a PR about this shortly.